### PR TITLE
Don't let a user connect multiple social users of the same network

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/social/SecureSocialUserServiceImpl.scala [hunk 1]
```diff
@@ -97,6 +97,12 @@ class SecureSocialUserPluginImpl @Inject() (
 
         //social user info with user must be FETCHED_USING_SELF, so setting user should trigger a pull
         //todo(eishay): send a direct fetch request
+
+        for (su <- socialUserInfoRepo.getByUser(user.id.get)
+            if su.networkType == socialUserInfo.networkType && su.id.get != socialUserInfo.id.get) {
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/social/SecureSocialUserServiceImpl.scala:102]: I think we should make a habit of filtering at the DB level. With Slick, it's not much extra code.

**Comment by 2is10** [comment]
Looks good to me.

File: kifi-backend/modules/shoebox/app/com/keepit/social/SecureSocialUserServiceImpl.scala (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/social/SecureSocialUserServiceImpl.scala:102]: Unless you're leveraging memcache here. I guess that's the exception. I kind of wish it were easy to tell at call sites whether a repo call leverages memcache.

Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/social/SecureSocialUserServiceImpl.scala:102]: We are leveraging memcache here. I agree, there should be an easier way to tell. Maybe some sort of naming convention?

**Pull request merged**